tests/tokenWrapper.test.ts

```diff
@@ -0,0 +1,800 @@
+        return api.disconnect()
+    });
+
+    before (async () => {
+        await api.isReady;
+    })
+
+    beforeEach(async () => {
+        ({ tokenWrapperContract, tokenWrapperContractFactory, sender, BobSigner, CharlieSigner, FerdieSigner, EveSigner, DaveSigner } =  await setup());
+    });
+
+    async function setup() {
+        const signerAddresses = await getAddresses();
+
+        const one = new BN(10).pow(new BN(api.registry.chainDecimals[0]))
+        const signers = await getSigners()
+
+        const Alice = signerAddresses[0];
+        const Bob = signerAddresses[1];
+        const Charlie = signerAddresses[2];
+        const Ferdie = signerAddresses[3];
+        const Eve = signerAddresses[4];
+        const Dave = signerAddresses[5];
+
+        const sender = await getRandomSigner(signers[0],one.muln(10));
+        const CharlieSigner = await getRandomSigner(signers[0],  one.muln(10));
+        const BobSigner = await getRandomSigner(signers[0], one.muln(10));
+        const FerdieSigner = await getRandomSigner(signers[0],  one.muln(10));
+        const EveSigner = await getRandomSigner(signers[0],  one.muln(10));
+        const DaveSigner = await getRandomSigner(signers[0], one.muln(10));
+
+        const {tokenName, tokenSymbol, decimal, contractGovernor, feeRecipient, feePercentage, isNativeAllowed, wrappingLimit,
+            contractProposalNonce, tokenAddress, totalSupply, governorBalance} = tokenWrapperContractInitParams(sender, BobSigner, CharlieSigner)
+
+
+        // token wrapper instantiation
+        const tokenWrapperContractFactory = await getContractFactory('governed_token_wrapper', sender.address);
+
+        const tokenWrapperContract = await tokenWrapperContractFactory.deploy('new',
+            tokenName, tokenSymbol, decimal, contractGovernor, feeRecipient, feePercentage, isNativeAllowed, wrappingLimit, contractProposalNonce, tokenAddress,
+            totalSupply, governorBalance);
+
+        return { sender, Alice, Charlie, CharlieSigner, Bob, BobSigner, Ferdie, FerdieSigner, Eve, EveSigner, Dave, DaveSigner,
+            tokenWrapperContractFactory, tokenWrapperContract};
+    }
+
+    function tokenWrapperContractInitParams(sender: any, BobSigner: any, CharlieSigner: any) {
+        let tokenName = 'Webb';
+        let tokenSymbol = 'Webb';
+        let decimal = 4;
+        let contractGovernor = sender.address;
+        let feeRecipient = sender.address;
+        let feePercentage = 1;
+        let isNativeAllowed = true;
+        let wrappingLimit = 10;
+        let contractProposalNonce = 0;
+        let tokenAddress = BobSigner.address
+        let totalSupply = 1_000_000_000_000_000;
+        let governorBalance = 9_000_000;
+
+
+        return {tokenName, tokenSymbol, decimal, contractGovernor, feeRecipient, feePercentage, isNativeAllowed, wrappingLimit,
+            contractProposalNonce, tokenAddress, totalSupply, governorBalance}
+    }
+
+    it('Add token address', async () => {
```

Comment:
This test seems to use `BobSigner.address` as the token address. This does not strike me as a clear and reasonable way to add a token. Is `BobSigner.address` a token? I would recommend creating a token using the PSP22 contracts and setting that to be the example address used. This would make the test a lot more clear and easy to follow as an outside observer.

Reply:
For this, I have been looking for a way to create PSP22 token and get the address of the token created since you gave your review, but I am not sure PSP22 supports/returns the address/account id of the token created. I have put up a question to the PSP22 team on Open brush for this.

Reply:
I think i can use the contract address as the token address for this